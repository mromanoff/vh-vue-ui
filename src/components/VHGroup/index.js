import VHGroup from './VHGroup.vue';

export default VHGroup;
