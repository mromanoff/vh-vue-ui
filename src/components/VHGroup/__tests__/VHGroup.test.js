import { shallowMount } from '@vue/test-utils';
import VHGroup from '../VHGroup.vue';

describe('VHGroup.vue', () => {
  let wrapper;

  it('renders default VHGroup', () => {
    wrapper = shallowMount(VHGroup);
    expect(wrapper.props().itemsPerRow).toBe(null);
    expect(wrapper.props().itemsPerRowTablet).toBe(null);
    expect(wrapper.props().itemsPerRowMobile).toBe(null);
    expect(wrapper.props().itemsPerRowDesktop).toBe(null);
    expect(wrapper.props().spaceBetweenItems).toBe(0);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders modified VHGroup', () => {
    wrapper = shallowMount(VHGroup, {
      propsData: {
        itemsPerRow: 4,
        itemsPerRowTablet: 4,
        itemsPerRowMobile: 2,
        itemsPerRowDesktop: 4,
        spaceBetweenItems: 2,
      },
    });
    expect(wrapper.props().itemsPerRow).toBe(4);
    expect(wrapper.props().itemsPerRowTablet).toBe(4);
    expect(wrapper.props().itemsPerRowMobile).toBe(2);
    expect(wrapper.props().itemsPerRowDesktop).toBe(4);
    expect(wrapper.props().spaceBetweenItems).toBe(2);
    expect(wrapper).toMatchSnapshot();
  });
});
