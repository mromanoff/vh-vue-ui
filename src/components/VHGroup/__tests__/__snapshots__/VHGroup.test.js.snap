// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VHGroup.vue renders default VHGroup 1`] = `<div class="Group" style="justify-items: stretch;"></div>`;

exports[`VHGroup.vue renders modified VHGroup 1`] = `<div class="Group Group-columnsMobile2--fraction Group-columnsTablet4--fraction Group-columnsDesktop4--fraction" style="grid-template-columns: repeat(4, 1fr); justify-items: stretch;"></div>`;
